pkg/operation/botanist/dns.go

```diff
@@ -369,3 +448,23 @@ func (b *Botanist) DeleteDNSProviders(ctx context.Context) error {
 		client.InNamespace(b.Shoot.SeedNamespace),
 	)
 }
+
+// MigrateInternalDNS destroys the internal DNSEntry, DNSOwner and DNSProvider resources
+// from the cluster, without removing it from the DNS provider
+func (b *Botanist) MigrateInternalDNS(ctx context.Context) error {
+	return component.OpDestroy(
+		b.Shoot.Components.Extensions.DNS.InternalOwner,
+		b.Shoot.Components.Extensions.DNS.InternalProvider,
+		b.Shoot.Components.Extensions.DNS.InternalEntry,
+	).Destroy(ctx)
+}
+
+// MigrateExternalDNS destroys the external DNSEntry, DNSOwner and DNSProvider resources
+// from the cluster, without removing it from the DNS provider
+func (b *Botanist) MigrateExternalDNS(ctx context.Context) error {
+	return component.OpDestroy(
+		b.Shoot.Components.Extensions.DNS.ExternalOwner,
+		b.Shoot.Components.Extensions.DNS.ExternalProvider,
+		b.Shoot.Components.Extensions.DNS.ExternalEntry,
+	).Destroy(ctx)
+}
```

Comment:
What does "without removing it from the DNS provider" mean? How does this work?

Reply:
Means that, it is not deleted in route53 for example and during the restoration, only the IP is updated. As @mandelsoft said the order of removing the DNS resources (Owner, Provider and Entry) is crucial for the dns controller, to achieve migration.

Reply:
OK, I see, thank you!